packages/frontend/services/azureFuncs.ts

```diff
@@ -19,22 +19,34 @@ export async function getProvenance(deviceKey: string) {
 export async function getAttachment(baseUrl: string, deviceKey: string, attachmentID: string) {
 //    const baseUrl = useRuntimeConfig().public.baseUrl;
     try {
-        const response = await fetch(`${baseUrl}/attachment/${deviceKey}/${attachmentID}`, {
-          method: "GET",
-        });
-        return await response.blob();
-      } catch (error) {
-        console.error('Error occurred during getAttachment request:', error);
-        throw error; // re-throw the error if you want to handle it further up the call stack
-      }
+    const response = await fetch(`${baseUrl}/attachment/${deviceKey}/${attachmentID}`, {
+      method: "GET",
+    });
+
+    const blob = await response.blob();
+
+    // Fetch the attachment name
+    const nameResponse = await fetch(`${baseUrl}/attachment/${deviceKey}/${attachmentID}/name`, {
```

Comment:
Should we check the response headers for `attachment-name` before calling into the `attachment/${deviceKey}/${attachmentID}/name` function? If deploying to the cloud resolves the CORS issue, we can avoid making two round trips every time.